### PR TITLE
feat: add IsDouble function

### PR DESCRIPTION
**Pull request by etiennep**
**Description:** Adds new a new `IsDouble` function to facilitate type checking. Most useful when checking the type of a body to determine if it needs to be parsed or not.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27895

**Testing:** Added unit test

**Documentation:** Updated the func readme.

### CODE REVIEW

**Review by codeboten — COMMENTED** [review]
Thanks @etiennep! Please sign the CLA

**Comment by atoulme** [comment]
Please run `make chlog-new` from the root of the repository to generate a changelog entry for this contribution, fill it up with issue information and add it to the PR.

File: pkg/ottl/ottlfuncs/README.md [hunk 1]
```diff
@@ -289,39 +289,56 @@ Unlike functions, they do not modify any input telemetry and always return a val
...
-- [Substring](#substring)
-- [Time](#time)
-- [TraceID](#traceid)
-- [TruncateTime](#truncatetime)
-- [UnixMicro](#unixmicro)
-- [UnixMilli](#unixmilli)
-- [UnixNano](#unixnano)
-- [UnixSeconds](#unixseconds)
-- [UUID](#UUID)
+- [OTTL Functions](#ottl-functions)
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/README.md:292]: @etiennep for this PR please only add the new Converter name to the list.  We can have a discussion about reorganizing these docs in a separate issue.

File: pkg/ottl/ottlfuncs/func_is_double_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		value    interface{}
+		expected bool
+	}{
+		{
+			name:     "float64",
+			value:    float64(2.7),
+			expected: true,
+		},
+		{
+			name:     "float64 without decimal",
```
Comment (TylerHelmuth) [inline pkg/ottl/ottlfuncs/func_is_double_test.go:28]: 👍

**Review by atoulme — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Comment by TylerHelmuth** [comment]
@etiennep I merged another function so we need a rebase.

**Comment by etiennep** [comment]
> @etiennep I merged another function so we need a rebase.

Done!

**Pull request merged**